zcash_client_sqlite/src/wallet/scanning.rs

```diff
@@ -1212,4 +1220,102 @@ mod tests {
             ]
         );
     }
+
+    #[test]
+    fn init_blocks_table_creates_scanned_range() {
+        use ScanPriority::*;
+
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), tests::network()).unwrap();
+        init_wallet_db(&mut db_data, Some(Secret::new(vec![]))).unwrap();
+
+        let sap_active = db_data
+            .params
+            .activation_height(NetworkUpgrade::Sapling)
+            .unwrap();
+        // Initialise the blocks table. We use Canopy activation as an arbitrary birthday height
+        // that's greater than Sapling activation.
+        let birthday_height = db_data
+            .params
+            .activation_height(NetworkUpgrade::Canopy)
+            .unwrap();
+        init_blocks_table(
+            &mut db_data,
+            birthday_height,
+            BlockHash([1; 32]),
+            1,
+            &[0x0, 0x0, 0x0],
+        )
+        .unwrap();
+
+        let expected = vec![
+            // The range below the wallet's birthday height is ignored
```

Comment:
This is not the range below the wallet's birthday height; it is the range up to and including the birthday height. Is that the intent?

Reply:
Yes; the birthday height is the height of the current chain tip at the time that the wallet is created; by definition, the wallet can receive no transactions within that block.

Reply:
```suggestion
            // The range up to and including the wallet's birthday height is ignored.
```